src/server/routes/contract/read/read-batch.ts

```diff
@@ -0,0 +1,170 @@
+import { Type, type Static } from "@sinclair/typebox";
+import type { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import SuperJSON from "superjson";
+import {
+  getContract,
+  prepareContractCall,
+  readContract,
+  resolveMethod,
+} from "thirdweb";
+import { prepareMethod } from "thirdweb/contract";
+import { resolvePromisedValue, type AbiFunction } from "thirdweb/utils";
+import { decodeAbiParameters } from "viem/utils";
+import { getChain } from "../../../../utils/chain";
+import { prettifyError } from "../../../../utils/error";
+import { thirdwebClient } from "../../../../utils/sdk";
+import { createCustomError } from "../../../middleware/error";
+import { standardResponseSchema } from "../../../schemas/sharedApiSchemas";
+import { getChainIdFromChain } from "../../../utils/chain";
+import { bigNumberReplacer } from "../../../utils/convertor";
+
+const MULTICALL3_ADDRESS = "0xcA11bde05977b3631167028862bE2a173976CA11";
+
+const MULTICALL3_AGGREGATE_ABI =
+  "function aggregate3((address target, bool allowFailure, bytes callData)[] calls) external payable returns ((bool success, bytes returnData)[])";
+
+const readCallRequestItemSchema = Type.Object({
+  contractAddress: Type.String(),
+  functionName: Type.String(),
+  functionAbi: Type.Optional(Type.String()),
+  args: Type.Optional(Type.Array(Type.Any())),
+});
+
+const readMulticallRequestSchema = Type.Object({
+  calls: Type.Array(readCallRequestItemSchema),
+  multicallAddress: Type.Optional(Type.String()),
+});
+
+const responseSchema = Type.Object({
+  results: Type.Array(
+    Type.Object({
+      success: Type.Boolean(),
+      result: Type.Any(),
+    }),
+  ),
+});
+
+const paramsSchema = Type.Object({
+  chain: Type.String(),
+});
+
+type RouteGeneric = {
+  Params: { chain: string };
+  Body: Static<typeof readMulticallRequestSchema>;
+  Reply: Static<typeof responseSchema>;
+};
+
+export async function readMulticall(fastify: FastifyInstance) {
+  fastify.route<RouteGeneric>({
+    method: "POST",
+    url: "/contract/:chain/read-batch",
+    schema: {
+      summary: "Batch read from multiple contracts",
+      description:
+        "Execute multiple contract read operations in a single call using Multicall3",
+      tags: ["Contract"],
+      operationId: "readMulticall",
```

Comment:
`readBatch` is preferred since "multicall" is the underlying approach but the use case is a batch read.